quickwit-cluster/src/cluster.rs

```diff
@@ -75,30 +87,64 @@ impl Member {
     }
 }
 
-impl TryFrom<NodeId> for Member {
-    type Error = anyhow::Error;
+impl From<Member> for NodeId {
+    fn from(member: Member) -> Self {
+        Self::new(member.internal_id(), member.gossip_public_address)
+    }
+}
 
-    fn try_from(node_id: NodeId) -> Result<Self, Self::Error> {
-        let (node_unique_id_str, generation_str) = node_id.id.split_once('/').ok_or_else(|| {
+// Build a cluster member from [`ClusterStateSnapshot`].
+// TODO: instead of passing a [`ClusterStateSnapshot`], passing a `NodeState` would be nicer but
+// the struct is not yet public.
+fn build_member_from_chichat<'a>(
+    node_id: &'a NodeId,
+    cluster_state_snapshot: &'a ClusterStateSnapshot,
+) -> anyhow::Result<Member> {
+    let (node_unique_id_str, generation_str) = node_id.id.split_once('/').ok_or_else(|| {
```

Comment:
`NodeId` should provide accessors for unique ID and generation.

Reply:
Actually, chitchat `NodeId` does not enforce that. This is the role of quickwit cluster to define the way to form an `internal_id`.

Ideally, I would like to have a `build` method on the member struct that takes a `node_id` and a `node_state`. Maybe having the `build` method on `Member` struct would be a bit better.

Reply:
I added a wrapper `QuickwitNodeId` to extract id & generation.